spyder/plugins/explorer/tests/test_explorer.py

```diff
@@ -46,5 +66,99 @@ def test_project_explorer(project_explorer):
+    project.explorer.treewidget.copy_path(fnames=file_paths,
+                                          method=path_method)
+    cb_output = cb.text(mode=cb.Clipboard)
+    file_paths = [_fn.replace(os.sep, "/") for _fn in file_paths]
+    if len(file_paths) > 1:
+        if path_method == 'absolute':
+            true_path = ''.join('"' + _fn + '",' + '\n' for _fn in file_paths)
+        elif path_method == 'relative':
+            true_path = ''.join('"' + osp.relpath(_fn, getcwd_or_home()).
+                                replace(os.sep, "/") + '",' +
+                                '\n' for _fn in file_paths)
+        true_path = true_path[:-2]
+    else:
+        if path_method == 'absolute':
+            true_path = file_paths[0]
+        elif path_method == 'relative':
+            true_path = (osp.relpath(file_paths[0], getcwd_or_home()).
+                         replace(os.sep, "/"))
+    assert true_path == cb_output
+
+
+@pytest.mark.parametrize('file_paths', [[project_file1],
+                                        [project_file1, project_file2, subdir],
+                                        [project_file1, project_file3]])
+def test_copy_file(copy_path_file, file_paths):
+    """Test copy file(s)/folders(s) to clipboard."""
+    project = copy_path_file
+    project.explorer.treewidget.copy_file_clipboard(fnames=file_paths)
+    cb_data = cb.mimeData().urls()
+    for url in cb_data:
+        file_name = url.toLocalFile()
+        assert osp.isdir(file_name) or osp.isfile(file_name)
+        if file_paths == [project_file1]:
+            with open(file_name, 'r') as fh:
+                text_data = fh.read()
+            assert text_data == 'Spyder4 will be released this year'
+        if file_paths == [project_file1, project_file3][1]:
+            with open(file_name, 'r') as fh:
+                text_data = fh.read()
+            assert text_data == 'South America'
+        if file_paths == [project_file1, project_file2, subdir][2]:
+            assert osp.isdir(file_name)
+
+
+@pytest.mark.parametrize('file_paths', [[subdir], [subdir, project_file3]])
+def test_save_file(copy_path_file, file_paths):
+    """Test save file(s)/folders(s) from clipboard."""
+    project = copy_path_file
+    project.explorer.treewidget.copy_file_clipboard(fnames=[project_file2])
+    project.explorer.treewidget.save_file_clipboard(fnames=file_paths)
+    assert osp.exists(osp.join(subdir, 'pyscript.py'))
+    with open(osp.join(subdir, 'pyscript.py'), 'r') as fh:
+        text_data = fh.read()
+    assert text_data == "Spyder4"
+    project.explorer.treewidget.copy_file_clipboard(fnames=[subdir])
+    project.explorer.treewidget.save_file_clipboard(fnames=[subdir,
+                                                            project_file3])
+    assert osp.exists(subdir + "1")
+    for afile in [project_file3, osp.join(subdir, 'pyscript.py')]:
+        assert osp.basename(afile) in os.listdir(subdir + '1')
+
+
+def test_delete_project_dir():
```

Comment:
This is not really a test, but rather a teardown routine for your ``copy_path_file`` fixture. Therefore, you should either make it a finalizer of the fixture, or easier just as teardown code after ``yield project_explorer`` in that fixture.

Reply:
Is this resolved now?